packages/components/src/components/select-native/select-native.jsx

```diff
@@ -62,111 +62,124 @@ const SelectNative = ({
     use_text,
     value,
     data_testid,
+    hide_placeholder,
     ...props
-}) => (
-    <div
-        className={classNames(className, 'dc-select-native', {
-            'dc-select-native--disabled': disabled,
-            'dc-select-native--error': error,
-            'dc-select-native--hide-selected-value': hide_selected_value,
-        })}
-    >
-        <div className='dc-select-native__wrapper'>
-            <div
-                className={classNames('dc-input', {
-                    'dc-input--disabled': disabled,
-                    'dc-input--error': error,
-                })}
-            >
-                <div className='dc-select-native__display'>
-                    {list_items && value && (
-                        <div
-                            className={classNames('dc-select-native__display-text', classNameDisplay)}
-                            data-testid='selected_value'
-                        >
-                            {!hide_selected_value && (use_text ? value : getDisplayText(list_items, value))}
-                        </div>
-                    )}
-                </div>
+}) => {
+    const [is_placeholder_hidden, setIsPlaceholderHidden] = React.useState(false);
+
+    React.useEffect(() => {
+        if (hide_placeholder && value) {
+            setIsPlaceholderHidden(true);
+        }
+    }, [value]);
+
+    return (
+        <div
+            className={classNames(className, 'dc-select-native', {
+                'dc-select-native--disabled': disabled,
+                'dc-select-native--error': error,
+                'dc-select-native--hide-selected-value': hide_selected_value,
+            })}
+        >
+            <div className='dc-select-native__wrapper'>
                 <div
-                    className={classNames('dc-select-native__placeholder', {
-                        'dc-select-native__placeholder--has-value': value,
-                        'dc-select-native__placeholder--disabled': disabled,
+                    className={classNames('dc-input', {
+                        'dc-input--disabled': disabled,
+                        'dc-input--error': error,
```

Comment:
Why are we updating `Component` class names?
These changes will affect all the instances of this component

Reply:
I suppose it was not updated, it was like this (deleted line 68), right?
@likhith-deriv @aizad-deriv I suggest removing useState & useEffect which are totally extra here and change line `103` to this:  `'dc-select-native__placeholder--hide_placeholder': hide_placeholder && value`. It's gonna work the same way 🙂